src/v/storage/batch_cache.cc

```diff
@@ -398,6 +450,28 @@ void batch_cache_index::truncate(model::offset offset) {
     }
 }
 
+void batch_cache_index::mark_clean(model::offset up_to) {
+    lock_guard lk(*this);
+
+    if (_dirty_tracker.clean() || up_to < _dirty_tracker.min()) {
```

Comment:
Should this be <=? It seems like if `up_to == _dirty_tracker.min()` it's a no-op

Reply:
Added a test and explanation for why it must be `<` https://github.com/redpanda-data/redpanda/pull/16954/files#diff-b4064461dbcccf747af446e1a4213b17661fc66447e7d4f4eb5f6061275aa3e8R323-R324

Reply:
Ah thanks. I was confused because I thought `up_to` was exclusive, but single-record batches makes a lot of sense. IIUC maybe this should be called `up_to_inclusive` or something?

>Marks batches below the specified offset as evictable.

I know this specifies batches below the offset, but I think it means "batches below or including the specified offset, provided the complete range is below the offset"